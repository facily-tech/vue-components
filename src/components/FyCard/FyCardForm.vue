<template>
  <fy-card elevation="0" class="fy-card-form pa-0 pa-md-4" v-bind="propsAndAttrs" v-on="$listeners">
    <v-card-title
      v-if="$slots['title']"
      class="fy-card-title text-h6 font-weight-medium justify-space-between"
    >
      <slot name="title" />
    </v-card-title>

    <v-card-subtitle v-if="$slots['subtitle']" class="fy-card-subtitle">
      <slot name="subtitle" />
    </v-card-subtitle>

    <slot v-for="(_, name) in $slots" :name="name" :slot="name" />
  </fy-card>
</template>

<script lang="ts">
import Vue from 'vue';
import FyCard from './FyCard.vue';

export default Vue.extend({
  name: 'FyCardForm',

  components: {
    FyCard,
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },
  },
});
</script>
