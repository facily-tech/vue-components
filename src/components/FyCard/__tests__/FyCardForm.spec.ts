import Vue from 'vue';
import Router from 'vue-router';
import FyCardForm from '../FyCardForm.vue';
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';
import Vuetify from 'vuetify';

Vue.use(Vuetify);

describe('FyCardForm', () => {
  // eslint-disable-next-line
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<any>;
  let router: Router;
  let localVue: typeof Vue;

  beforeEach(() => {
    router = new Router();
    localVue = createLocalVue();
    localVue.use(Router);

    mountFunction = (options = {}) => {
      return mount(FyCardForm, {
        localVue,
        vuetify: new Vuetify(),
        router,
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    expect(mountFunction().html()).toMatchSnapshot();
  });
});
