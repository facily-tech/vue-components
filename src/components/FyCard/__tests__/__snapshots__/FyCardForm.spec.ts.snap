// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyCardForm should render component and match snapshot 1`] = `<div class="fy-card-form pa-0 pa-md-4 v-card v-sheet theme--light elevation-0"></div>`;
